### ISSUE TITLE
customHTMLFiles explanation

### ISSUE THREAD

**Issue by taohe009** [Stirling-Tools/Stirling-PDF#1892]
```
customHTMLFiles: false # enable to have files placed in /customFiles/templates override the existing template html files
```

Can you give me a detailed explanation of how to mount after this function is enabled? I failed to mount it and cannot customize it. How can I add other custom codes to the homepage?

**Comment by Frooodle** [comment]
`` I failed to mount it and cannot customize it.``
can you expand what you mean by this?
try just do a docker mount on /customFiles as per our examples

**Comment by taohe009** [comment]
I mean, how do I add ad code to the homepage, how do I modify the homepage? Please teach me how.

**Comment by sbplat** [comment]
Set it to `true` and place your HTML files in `/customFiles/templates`. You can view the default files [here](https://github.com/Stirling-Tools/Stirling-PDF/tree/main/src/main/resources/templates).

**Comment by taohe009** [comment]
> 将其设置为 并将 HTML 文件置入 。您可以[在此处](https://github.com/Stirling-Tools/Stirling-PDF/tree/main/src/main/resources/templates)查看默认文件。`true``/customFiles/templates`

src/main/resources
/templates/
I need to download the files in this file and upload them to my server, right?

**Comment by Frooodle** [comment]
Sbplat linked them in his above comment !

**Comment by taohe009** [comment]
> Sbplat 在他上面的评论中链接了他们！

I used docker exec -it stirling-pdf /bin/bash to enter the container. After entering the container, I couldn't find the homepage I wanted to modify. How can I modify the running container file?

**Comment by taohe009** [comment]
docker-compose.yml  Is it correct to change the file to the following?

services:
  stirling-pdf:
    image: frooodle/s-pdf:latest
    volumes:
      - /path/to/trainingData:/usr/share/tessdata
      - /path/to/stirling-pdf/src/main/resources/templates:/customFiles/templates  # 添加自定义文件的卷挂载
    environment:
      - SYSTEM_CUSTOMHTMLFILES=true  # 启用自定义HTML文件支持

**Comment by taohe009** [comment]
docker-compose.yml Is it correct to change the file to the following?

services:
stirling-pdf:
image: frooodle/s-pdf:latest
volumes:
- /path/to/trainingData:/usr/share/tessdata
- /path/to/stirling-pdf/src/main/resources/templates:/templates
environment:
- SYSTEM_CUSTOMHTMLFILES=true

**Comment by Frooodle** [comment]
/customFiles/templates on docker side

**Comment by taohe009** [comment]
- /root/Stirling-PDF/customFiles/templates:/templates

Does this work?

**Comment by Frooodle** [comment]
remote path MUST be ``/customFiles/templates``

**Comment by taohe009** [comment]
I am a newbie and I don't know how to use Docker before. I used to build websites with PHP or create HTML on WWW. Can you explain it in detail? Please, I don't know how to add code to the homepage of the website deployed by Docker, or modify the homepage, thank you.

**Comment by taohe009** [comment]
Is this the correct setting?

version: '3.3'
services:
  stirling-pdf:
    image: frooodle/s-pdf:latest
    ports:
      - '8080:8080'
    volumes:
      - ./path/to/trainingData:/usr/share/tessdata #Required for extra OCR languages
      - ./extraConfigs:/configs
#      - ./customFiles:/customFiles/
#      - ./logs:/logs/
    environment:
      - DOCKER_ENABLE_SECURITY=false
      - INSTALL_BOOK_AND_ADVANCED_HTML_OPS=false
      - LANGS=en_GB

**Comment by taohe009** [comment]
![image](https://github.com/user-attachments/assets/a3b8c17b-c480-46c7-844a-66aee3da1c0a)

**Comment by Frooodle** [comment]
nnow if you edit those files it should show on application (cant remember if it needs restart)

**Comment by taohe009** [comment]
> nnow if you edit those files it should show on application (cant remember if it needs restart)

Can I only change the home.html file for the home page?

**Comment by taohe009** [comment]
No, no matter how I change it, the homepage doesn't change, and the official link at the bottom of the homepage is gone, which is very strange.

https://pdfgoto.com

**Comment by taohe009** [comment]
Could you help me check if my steps are correct?

1. Enter
docker exec -it stirling-pdf /bin/bash

2. Install Git using the package manager:
apk update
apk add git

3. Clone a specific folder: Use the git clone command to clone a specific folder:
git clone --depth 1 --filter=blob:none --sparse https://github.com/Stirling-Tools/Stirling-PDF.git
cd Stirling-PDF
git sparse-checkout init --cone
git sparse-checkout set src/main/resources/templates

4. Copy files: Copy the cloned files to the /customFiles/templates directory.
cp -r src/main/resources/templates/* /customFiles/templates/

5. Enter the directory:
cd /customFiles/templates

6. View the directory:
ls /customFiles/templates

7. Edit the file:
nano /customFiles/templates/home.html

After editing, press Ctrl + X to exit the editor, press Y to save the changes, and press Enter to confirm the file name.

**Comment by Frooodle** [comment]
Seems correct

**Comment by taohe009** [comment]
I changed the Stirling hyperlink at the bottom of the homepage to a hyperlink to my website. After restarting the container, the link at the bottom disappeared, https://pdfgoto.com/

**Comment by taohe009** [comment]
> 似乎是正确的
nano /customFiles/templates/home.html
I changed the

https://stirlingpdf.info/s/clwzgtfw7000gltkmwz1n212m in home.html to my https://www.pdfgoto.com/. After restarting the container, I opened the website and the link was still: https://stirlingpdf.info/s/clwzgtfw7000gltkmwz1n212m

Nothing changed???

**Comment by taohe009** [comment]
docker run -d \
-p 8080:8080 \
-v /location/of/trainingData:/usr/share/tessdata
-v ./extraConfigs:/configs \
-v ./logs:/logs \
-v /location/of/customFiles:/customFiles \
-e DOCKER_ENABLE_SECURITY=false \
-e INSTALL_BOOK_AND_ADVANCED_HTML_OPS=false \
-e LANGS=en_GB \
--name stirling-pdf \
frooodle/s-pdf:latest

Please check if the command I run in this container is correct?

**Comment by taohe009** [comment]
docker run -d \
-p 8080:8080 \
-v /location/of/ trainingData :/ usr /share/ tessdata
- v ./extraConfigs : / configs\
-v ./logs:/logs \
- v . /location/of/ customFiles :/ customFiles \
-e DOCKER_ENABLE_SECURITY=false \
-e INSTALL_BOOK_AND_ADVANCED_HTML_OPS=false \
-e LANGS= en_GB \
--name stirling -pdf \
frooodle / s- pdf:latest

Please check if the command I run in this container is correct? Or is it like the following?

docker run -d \
-p 8080:8080 \
-v /location/of/ trainingData :/ usr /share/ tessdata
- v ./extraConfigs : / configs\
-v ./logs:/logs \
- v ./customFiles:/customFiles/
-e DOCKER_ENABLE_SECURITY=false \
-e INSTALL_BOOK_AND_ADVANCED_HTML_OPS=false \
-e LANGS= en_GB \
--name stirling -pdf \
frooodle / s- pdf:latest

Like this? Or like this? :

docker run -d \
-p 8080:8080 \
-v /location/of/ trainingData :/ usr /share/ tessdata
- v ./extraConfigs : / configs\
-v ./logs:/logs \
- v ./customFiles / templates: / customFiles /
-e DOCKER_ENABLE_SECURITY=false \
-e INSTALL_BOOK_AND_ADVANCED_HTML_OPS=false \
-e LANGS= en_GB \
--name stirling -pdf \
frooodle / s- pdf:latest

Or like this :

docker run -d \
-p 8080:8080 \
-v /location/of/ trainingData :/ usr /share/ tessdata
- v ./extraConfigs : / configs\
-v ./logs:/logs \
/ customFiles /templates:/templates
-e DOCKER_ENABLE_SECURITY=false \
-e INSTALL_BOOK_AND_ADVANCED_HTML_OPS=false \
-e LANGS= en_GB \
--name stirling -pdf \
frooodle / s- pdf:latest

**Comment by taohe009** [comment]
？

**Comment by Frooodle** [comment]
-v ./trainingData :/usr/share/tessdata
-v ./extraConfigs : /configs
-v ./logs:/logs \
v ./customFiles:/customFiles
